apps/forms/form-dialog-alert/src/app/guards/unload-window.directive.ts

```diff
@@ -0,0 +1,13 @@
+import { Directive, HostListener, inject } from '@angular/core';
+import { Observable } from 'rxjs';
+import { FormComponent } from '../ui/form.component';
+
+@Directive({ selector: 'app-form[appUnloadGuard]', standalone: true })
+export class UnloadGuardDirective {
+  readonly component = inject(FormComponent);
+
+  @HostListener('window:beforeunload')
+  canDeactivate(): Observable<boolean> | boolean {
+    return !this.component.form!.dirty;
```

Comment:
This gives `Forbidden non null assertion` eslint error.   You can replace it with `!this.component.form?.dirty`.